drivers/hdc1000/hdc1000.c

```diff
@@ -111,19 +119,42 @@ int hdc1000_get_results(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
 
     if (status == HDC1000_OK) {
         /* if all ok, we convert the values to their physical representation */
+        uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
+        temp_cached = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
```

Comment:
This seems conceptual wise broken to me: why should the `get_results()` care about caching? All the caching should be handled in the `hdc1000_read_cached function()` and the changes should be non-intrusive for the existing functions!